### PR TITLE
Jessica's Assignment 6

### PR DESCRIPTION
**Pull request by jessmee**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -25,5 +25,7 @@ <h1>CitiPix</h1>
 					<input type="submit" value="Update" id="submit-btn">
 				</form>
 			</div>
+		<script src="http://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+    <script src="js/script.js"></script>
```
Comment (ericmasiello) [inline index.html:29]: Watch your indentation. These two `<script>` elements should be indented to the same level.

File: js/script.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  alert('Oops! There is no picture for this city!');
+}
+
+function findCity(event) {
+  event.preventDefault(); 
+  console.log('submitting the form');
+
+  var city = $('#city-type').val();
+  
+    if (city.toLowerCase().toUpperCase() === 'New York'.toLowerCase().toUpperCase() || city.toLowerCase().toUpperCase() === 'New York City'.toLowerCase().toUpperCase() || city.toLowerCase().toUpperCase() === 'NY'.toLowerCase().toUpperCase() || city.toLowerCase().toUpperCase() === 'NYC'.toLowerCase().toUpperCase()) {
```
Comment (ericmasiello) [inline js/script.js]: Right idea. Although, there's no need to convert to lowercase and then to uppercase. Just do one or the other.

You also could have saved yourself a bit of typing if you had "normalized" the `city` variable prior to the `if` block.

e.g.

```js
var city = $('#city-type').val();
city = city.toLowerCase(); // now city is always lower cased

// now compare city to the lower case variants of each city
if (city === 'new york city' || city === 'new york' || city === 'nyc' || city === 'ny') {
  // ....
}
```

File: js/script.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      $('body').removeClass();
+      $('body').addClass('nyc');
```
Comment (ericmasiello) [inline js/script.js]: Nice! One other thing you can do is leverage chaining:

```js
$('body').removeClass().addClass('nyc');
```

File: js/script.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      $('body').addClass('la');           
+    } else if (city.toLowerCase().toUpperCase() === 'Austin'.toLowerCase().toUpperCase() || city.toLowerCase().toUpperCase() === 'ATX'.toLowerCase().toUpperCase()) {
+      $('body').removeClass();
+      $('body').addClass('austin');
+    } else if (city.toLowerCase().toUpperCase() === 'Sydney'.toLowerCase().toUpperCase() || city.toLowerCase().toUpperCase() === 'SYD'.toLowerCase().toUpperCase()) {
+      $('body').removeClass();
+      $('body').addClass('sydney');      
+    } else {
+      $('body').removeClass();
+      return error(); //using an alert box, but how can I make a popup?
```
Comment (ericmasiello) [inline js/script.js:59]: Whatt kind popup are you looking to build here?

File: js/script.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+console.log('hello is this showing up?');
+//The user types in the name of the city
+//The user clicks on the submit button, which carries out the function findCity
+//The function is prevented from going into the default action
+//The function extracts and stores the user input
+//the user may use lowercase, so make sure strings are converted to lowercase.
+//The stored name of the city is then evaluated to a set of conditionals
+function error() {
+  alert('Oops! There is no picture for this city!');
```
Comment (ericmasiello) [inline js/script.js:9]: Nice!

File: js/script.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  } else if (city === 'los angeles' || city === 'la' || city === 'lax') {
+    $('body').removeClass().addClass('la');           
+  } else if (city === 'austin' || city === 'atx') {
+    $('body').removeClass().addClass('austin');
+  } else if (city === 'sydney' || city === 'syd') {
+    $('body').removeClass().addClass('sydney');      
+  } else {
+    $('body').removeClass();
+    return error(); //using an alert box, but how can I make a popup?
+    event.default();
```
Comment (ericmasiello) [inline js/script.js:32]: You don't need this line for two reasons.

1. You've already called `event.preventDefault()` on line 13.
2. Since you have a `return` statement before this on line 31, it will actually never even hit line 32.

File: js/script.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  } else if (city === 'san francisco' || city === 'san fran' || city === 'bay area' || city === 'sf') {
+    $('body').removeClass().addClass('sf');           
+  } else if (city === 'los angeles' || city === 'la' || city === 'lax') {
+    $('body').removeClass().addClass('la');           
+  } else if (city === 'austin' || city === 'atx') {
+    $('body').removeClass().addClass('austin');
+  } else if (city === 'sydney' || city === 'syd') {
+    $('body').removeClass().addClass('sydney');      
+  } else {
+    $('body').removeClass();
```
Comment (ericmasiello) [inline js/script.js:30]: Since in every single case you're calling `$('body').removeClass()`, you can DRY (**d**on't **r**epeat **y**ourself) up your code by calling `$('body').removeClass();` before your entire `if` block.

e.g.

```js
// ...
city = city.toLowerCase();
$('body').removeClass();

if (city === 'new york' || city === 'new york city'|| city === 'ny' || city === 'nyc') {
  $('body').addClass('nyc');
} else if (/* ... */) {
  $('body').addClass('sf'); 
} // etc...
```
In this code structure, you wouldn't even need the `else` since you already handled `$('body').removeClass();` before the `if` block.

Comment (jessmee) [reply js/script.js:30]: If I use this structure , how can I still get the alert box without using else?

Comment (ericmasiello) [reply js/script.js:30]: Ah excellent point! So you would in fact need the `else` but now you would only need to call `return error();`
